libretroshare/src/dht/p3bitdht_relay.cc

```diff
@@ -322,7 +322,7 @@ bool    p3BitDht::loadList(std::list<RsItem *>& load)
 		}
 		else if (0 == strncmp(key.c_str(), "RELAY_MODE", 10))
 		{
-			mode = atoi(value.c_str());
+			mode = static_cast<RsDhtRelayMode>(atoi(value.c_str()));
```

Comment:
you need to check that atoi returns a value that is in the range of existing values. Otherwise it's possible to send crafted items that would kill your RS node.

Reply:
This is loaded from your configuration files, that should be rather safe (`p3BitDht::loadList`).